src/_pytest/junitxml.py

```diff
@@ -252,7 +252,10 @@ def append_error(self, report):
 
     def append_skipped(self, report):
         if hasattr(report, "wasxfail"):
-            self._add_simple(Junit.skipped, "expected test failure", report.wasxfail)
+            xfailreason = report.wasxfail
+            if xfailreason.startswith("reason: "):
+                xfailreason = xfailreason[8:]
+            self._add_simple(Junit.skipped, xfailreason, report.wasxfail)
```

Comment:
Shouldn't it also use `bin_xml_escape` like with the block below?

From the issue currently it would look like this:

> <skipped message="expected test failure">reason: lets see XFAIL</skipped>

While this changes it to this then, correct?

> <skipped message="lets see XFAIL">reason: lets see XFAIL</skipped>

Skips appear to look like this, i.e. they have a type, and no element contents (I assume this comes via `report.wasxfail` as the 3rd arg still):

>  <skipped message="lets see SKIP" type="pytest.skip">

Should/could we also have a type here?  (`pytest.xfail`)

Reply:
Good point - I had overlooked that the `report.xfailed` was still being passed as the content. I have modified this PR to use the same `Junit.skipped` method to add the type.

Reply:
Great!